silencer-plugin/src/main/scala/com/github/ghik/silencer/SuppressingReporter.scala

```diff
@@ -31,7 +31,7 @@ class SuppressingReporter(original: Reporter, globalFilters: List[Regex]) extend
     severity match {
       case INFO =>
         original.info(pos, msg, force)
-      case WARNING if globalFilters.exists(_.findFirstIn(msg).isDefined) =>
+      case WARNING if existsIn(globalPathFilters, pos.source.path) || existsIn(globalFilters, msg) =>
```

Comment:
`pos.source.path` is (or may be) absolute which means it may be different on each machine which may result in false positives and non-deterministic builds in general. Ideally these paths should be relative to project root, but `scalac` doesn't have access to it. However, we might introduce another option for passing source root directory (or directories). Paths matched by regexes would be then relativized with respect to these roots.

Also, I wonder what should we do with different path separators on different operating systems (`/` vs `\`). IMHO, we should normalize the path to use `/`.

Reply:
I agree on the introduction of the option to specify a source root directory be it regex or not, which we could then remove from the source path before we apply the regex specified in the `globalPathFilters`.

However, I disagree on the standardizing on the UNIX file separator as it is counterintuitive. Having the possibility to introduce multiple regexes that handle the file separators of the file systems you are executing the code on IMHO falls under the responsability of the user.

Reply:
@mihaisoloi What I care about is that builds are reproducible, i.e. work exactly the same on every environment. You should never assume what system your project is being built on. You may use e.g. Windows and specify path patterns containing `\` but the moment you put your project on Travis CI or someone using Unix clones it, it will no longer work. Having to write "cross-platform" regexes also doesn't look like an elegant solution to me.
Yes, you are right that normalizing filename separators to `/` may be confusing for a Windows user but IMHO non-reproducible builds are *much* worse than that. I think it should be enough to do this normalization and document it well and explicitly.

As for the root directories - I want them for exactly the same reason - build reproducibility. I want files to have paths determined by project structure *only* - it must not matter where the project is being checked out in the file system and on which operating system.
I don't see a reason for root directories to be specified as regexes. This seems like an unnecessary complexity for me. Normally, these dirs are easy to pass from `sbt`, e.g. from `baseDirectory` or `sourceDirectories` keys as literal, absolute paths. I think this is what's going to happen 95% of the time so I don't see a reason to make users remember to additionally escape these paths into regexes.

Reply:
I agree with all your points. Builds must be reproducible regardless of environment, I applied the changes you suggested. Should I also bump the minor or patch version?

Reply:
Thanks! I'll take care of version management and release when the PR is merged.